app/core/src/screens/hotels/AllHotelsMapNavigationScreen.js

```diff
@@ -0,0 +1,46 @@
+// @flow
+
+import * as React from 'react';
+import { connect } from 'react-redux';
+import {
+  AllHotelsMap,
+  type SearchParametersType,
+} from '@kiwicom/react-native-app-hotels';
+
+import type { Navigation } from '../../types/Navigation';
+
+type Props = {|
+  navigation: Navigation,
+  hotelsFilter: SearchParametersType,
+  onFilterChange: Object => void,
```

Comment:
I miss stricter type control here. There is no check on return of `mapStateToProps`, `mapDispatchToProps` and no check on props of the exported container.

I would suggest rewriting this type into an intersection of three types for each source of props:

```js
type ContainerProps = {|
  navigation: Navigation,
|};

type StateProps = {|
  hotelsFilter: SearchParametersType,
|};

type DispatchProps = {|
  onFilterChange: Object => void,
|};

type Props = ContainerProps & StateProps & DispatchProps;
```

Mapping functions would then change from typeless functions to `(state: ReduxState): StateProps =>` and `(dispatch): DispatchProps =>` and the exported container could be typed as `React.ComponentType<ContainerProps>`.

Reply:
And there should be probably `SearchParametersType` instead of `Object`.

Reply:
Thanks, this is an absolutely fantastic comment! 💯